File: javascript/003_largest_prime_factor.js
function largestPrimeFactor(targetNumber) {
  let number = targetNumber;
  let divisor = 2;
  let largest = 0;

  while (number != 1) {
    if (number % divisor === 0) {
      number /= divisor;
      (largest < divisor) ? largest = divisor : largest;
    } else {
      divisor += 1;
    }
  }
  return largest;
}

console.log(largestPrimeFactor(600851475143));

File: c/003_largest_prime_factor.c
#include <stdio.h>

int	largest_prime_factor(long int number)
{
	int	divisor;
	int	largest;

	divisor = 2;
	largest = 0;
	if (number > 1)
	{
		while (number != 1)
		{
			if (number % divisor == 0)
			{
				number = number / divisor;
				largest = divisor;
			}
			else
			{
				divisor++;
			}
		}
	}
	return (largest);
}

int	main(void)
{
	printf("%d\n", largest_prime_factor(600851475143));
	return (0);
}
